src/sbvr-api/abstract-sql.ts

```diff
@@ -140,14 +141,32 @@ export const getAndCheckBindValues = async (
 			}
 
 			try {
-				return await AbstractSQLCompiler[engine].dataTypeValidate(value, field);
+				return await validateBindingType(engine, value, field);
 			} catch (err: any) {
 				throw new BadRequestError(`"${fieldName}" ${err.message}`);
 			}
 		}),
 	);
 };
 
+const validateBindingType = async (
+	engine: Engines,
+	$value: any,
+	field: { dataType: string },
+): Promise<any> => {
```

Comment:
Is `any` the correct type here? I assume it probably is but want to check

Reply:
Yes, it is the incoming type from abstract-sql-compiler

```
dataTypeValidate: (value: any, field: Pick<AbstractSqlField, "dataType" | "required">) => Promise<any>;
```

I initially tried using `ReturnType<Awaited<typeof AbstractSQLCompiler[engine]dataTypeValidate>>`  but the dynamic `engine` break this typing